ts/components/messages/paginated/MessageDetail/MedicalPrescriptionAttachments.tsx

```diff
@@ -0,0 +1,117 @@
+
+import { PrescriptionData } from "../../../../../definitions/backend/PrescriptionData";
+import I18n from "../../../../i18n";
+import customVariables from "../../../../theme/variables";
+import { getPrescriptionDataFromName } from "../../../../utils/messages";
+import { Attachment } from "../../../../store/reducers/entities/messages/types";
+import ItemSeparatorComponent from "../../../ItemSeparatorComponent";
+
+type Props = Readonly<{
+  prescriptionData?: PrescriptionData;
+  attachments: ReadonlyArray<Attachment>;
+  organizationName?: string;
+}>;
+
+const BARCODE_HEIGHT = 52;
+
+const styles = StyleSheet.create({
+  padded: {
+    paddingHorizontal: customVariables.contentPadding
+  },
+  image: {
+    width: 300,
+    height: BARCODE_HEIGHT,
+    resizeMode: "contain"
+  },
+  note: {
+    lineHeight: 16
+  },
+  label: {
+    lineHeight: 22
+  },
+  customHeader: {
+    marginBottom: -4
+  }
+});
+
+export const svgXml = "image/svg+xml";
+
+const footerItem = () => (
+  <React.Fragment>
+    <ItemSeparatorComponent />
+    <View spacer={true} />
+    <Text style={[styles.note, styles.padded]}>
+      {I18n.t("messages.medical.note")}
+    </Text>
+  </React.Fragment>
+);
+
+const headerItem = (organizationName?: string) => (
+  <View style={styles.padded}>
+    <Text bold={true} style={styles.customHeader}>
+      {I18n.t("messages.medical.nationalService").toUpperCase()}
+    </Text>
+    {organizationName && (
+      <Text style={styles.label}>{organizationName.toUpperCase()}</Text>
+    )}
+    <View spacer={true} xsmall={true} />
+    <ItemSeparatorComponent noPadded={true} bold={true} />
+  </View>
+);
+
+const renderItem =
+  (
+    attachmentsWithSvg: Array<Attachment>,
+    prescriptionData?: PrescriptionData
+  ) =>
+  ({ item }: { item: Attachment }) => {
+    const value = getPrescriptionDataFromName(prescriptionData, item.name);
+    const xml = fromNullable(attachmentsWithSvg.find(_ => _.name === item.name))
+      .map(({ content }) => Buffer.from(content, "base64").toString("ascii"))
+      .toUndefined();
+    return (
+      <View style={styles.padded}>
+        <View spacer={true} small={true} />
+        <Text style={styles.label}>
+          {I18n.t(`messages.medical.${item.name}`, {
+            defaultValue: I18n.t("messages.medical.not_available")
+          }).toUpperCase()}
+        </Text>
+        {xml && <SvgXml xml={xml} width={"100%"} height={BARCODE_HEIGHT} />}
+        {value.isSome() && (
+          <Text semibold={true} style={{ textAlign: "center" }}>
+            {I18n.t("global.symbols.asterisk")}
+            {value.value}
+            {I18n.t("global.symbols.asterisk")}
+          </Text>
+        )}
+        <View spacer={true} />
+      </View>
+    );
+  };
+
+const MedicalPrescriptionAttachments = ({
+  attachments,
+  organizationName,
+  prescriptionData
+}: Props) => {
+  const attachmentsWithSvg = attachments.filter(_ => _.mimeType === svgXml);
+
+  return (
+    <FlatList
```

Comment:
This component is quite hard for me to understand because we have no real testing in place.
I translated it blindly from the legacy one where I noticed that we suffer from the very same issues you are pointing out here.

That's why I first [updated the dev server](https://github.com/pagopa/io-dev-api-server/pull/98/files#diff-c4cdb7cf0e69ecfc696d0ba80b8411dd4297658be4a57dad5f0954d7a9ae6d74R147) to send a realistic payload, thus having a "control group" for my refactoring. And here we are:

![Screenshot 2021-11-29 at 18 19 35](https://user-images.githubusercontent.com/2094604/143925741-2eef00ee-1f98-4dbd-8410-bafad7293d2e.png)

Refactored code [here](https://github.com/pagopa/io-app/pull/3519/files#diff-4d94a160c2c2878851018c8cf7b1997791f569c9901a48a45dcde5b82ab03ddcR82).